client/task-list/tasks/payments/PaymentGatewaySuggestions/index.js

```diff
@@ -51,21 +50,39 @@ export const PaymentGatewaySuggestions = ( { query } ) => {
 			.getPaymentGatewaySuggestions()
 			.reduce( ( map, suggestion ) => {
 				const { id } = suggestion;
-				map.set( id, suggestion );
+				const gateway = gateways[ id ] ? gateways[ id ] : {};
+				const enrichedSuggestion = {
+					installed: !! gateways[ id ],
+					postInstallScripts: gateway.post_install_scripts,
+					enabled: gateway.enabled,
+					needsSetup: gateway.needs_setup,
+					requiredKeys: gateway.required_settings_keys,
+					settingsUrl: gateway.settings_url,
+					connectionUrl: gateway.connection_url,
+					setupHelpText: gateway.setup_help_text,
+					settings: gateway.settings,
```

Comment:
We could probably just pass `requiredSettings` avoiding the need to pass `requiredKeys` and `settings` in the child component.

Reply:
Good call 💯

Reply:
Done in 2d60fac5d509fc78e59d77f2e4e27cdb210c26c7